components/server/src/user/user-service.ts

```diff
@@ -312,3 +313,53 @@ export class UserService {
         log.info("User verified", { userId: user.id });
     }
 }
+
+// TODO: refactor where this is referenced so it's more clearly tied to just analytics-tracking
+// Let other places rely on the ProfileDetails type since that's what we store
```

Comment:
need to remove this comment 🙈 
but that describes exactly what was done here, the flat shape for analytics is not to be used on dashboard, as we already have ProfileDetails. especially, during the actual task of adapting new API this broke silently, as ProfileDetails and this one are not 100% compatible.